### ISSUE TITLE
v14.0.1 Release Status - Beta Download

### ISSUE THREAD

**Issue by cleidigh** [thundernest/import-export-tools-ng#466]
Latest beta : 
https://github.com/thundernest/import-export-tools-ng/blob/v14.0.1/xpi/beta/import-export-tools-ng-14.0.1-b1-tb.xpi

Status: 

b1
- fix csv output to text not HTML 
- #451 , #463 
-

**Comment by Yarny0** [comment]
Thanks for the quick implementation, @cleidigh.

I tested most export functions and formats, and everything worked as I assume it should work, with these exceptions:

*   Exporting a complete mail account just generates an empty directory.
    I tested this with the context menu of the account's entry in the folder pane, where it says "Export Accont - (ACCOUNT-NAME)".
    The Error Console logs the following message:

    ```
    Uncaught (in promise) Error: Component returned failure code: 0x80550006 [nsIMsgFolder.messages]    mboxImportExport.js:507
      buildAndExportMbox chrome://mboximport/content/mboximport/modules/mboxImportExport.js:507
      exportFoldersToMbox chrome://mboximport/content/mboximport/modules/mboxImportExport.js:455
      exportAccount chrome://mboximport/content/mboximport/mboximport.js:1002
    ```

*   If both options "Use .mbox extension for mbox files" and "Use just alphanumeric characters in export" are enabled, exporting a folder as "Single Zipped mbox File" generates an incomplete zip file (it's just 22 Bytes in size, it's probably just a zip header).
    The Error Console logs the following message:

    ```
    10:29:08.910 Uncaught (in promise) NS_ERROR_FILE_NOT_FOUND: Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIFile.isDirectory]     mboximport.js:855
      IETexportZip chrome://mboximport/content/mboximport/mboximport.js:855
    ```

    It also generates a file `(FOLDER-NAME)_mbox` in the same directory, which contains the unzipped mbox.  The export process maybe gets interrupted before it can zip and then remove the unzipped file.

For these tests I used Thunderbird 115.3.1 (on NixOS 23.05) with a fresh profile, with all settings at their default and only ITE (14.0.1-b1) installed as addon.  I created a POP3-connected mail account with two small messages in its Inbox to test exports.

(Note: I'm not relying on mbox or full account export functions.  My users mainly need IET for EML-export of individual messages.  But I'd like to help development with testing the beta version.  I hope I'm helping with this report.  Thanks for IET!)

**Comment by DenB10** [comment]
Hi,
I can't confirm the first comment, but I can confirm the second.
Everything else works fine for me except the French translation, which has not been fully updated (in the _locales folder and the help file). In addition, the clipboard copy warning window should be translated when several messages are selected!
Best regards,
@DenB10

**Comment by cleidigh** [comment]
@DenB10 
Hope your vacation was good!
I think I must have merged the wrong zip from you, I will verify and remerge.
On the clipboard warning, that actually is not translated yet (still static English) so neither of us missed that. Will do when I have the next batch.
Thanks again for the effort, vive la France %-)
@cleidigh

**Comment by cleidigh** [comment]
@Yarny0 
Glad the help options get you back into your workflow.
Your report and interest in helping the development beyond even what you utilize is so helpful!!

@DenB10 
@Yarny0 
ImportExportTools is a really difficult tool because it not only has to deal with the ever changing Thunderbird (v115 required really major changes). It also has to deal with many external data that most of the time I cannot see.
I am but one, disabled person so I do my best, however, its help from people like yourselves that makes this possible.
Thanks again!
Christopher

**Comment by cleidigh** [comment]
@Yarny0 
@DenB10 
On the account export : 
I had one earlier report. This is a fundamental failure where I am asking for the message iterator for each folder to export one by one. The process completely fails because Thunderbird throws an error. This is something that I am worried will be tricky to figure out.

@DenB10 
@Yarny0 
Were both of your accounts pop type where you saw this failure?

The second issue should be a basic fix.
@cleidigh

**Comment by DenB10** [comment]
@cleidigh

>Hope your vacation was good!

My vacation went well, thank you! Except that it was really too crowded where I was (Florence in Italy). Mostly insufferable people whose sole aim was to take a photo of themselves in front of "Primavera" from Botticelli with a thumbs-up. I would have liked to answer them with my middle finger raised if I had their education...

>vive la France %-)

I'm not patriotic because the idea of homeland is a pretext for war. I'd rather be a citizen of the world, and when everyone is (in vain hope), we'll see who the real enemies are.

But back to IET NG:

>I think I must have merged the wrong zip from you, I will verify and remerge.

Don't worry, I'll propose changes directly in the v14.0.1 branch by pull request if you prefer.
Especially as the translation will evolve with the future changes.

>Were both of your accounts pop type where you saw this failure?

My accounts are all imap accounts and I don't have the first problem Yarny0 mentioned with exporting an account. I simply have the same remark (second point) that he indicates for exporting a folder.

Last point I would like to make:JI I think that "msgDisplayCtxMenuSet" is no longer needed in the "wextMenus.js" file because it's redundant. What do you think?

**Comment by cleidigh** [comment]
@DenB10 
Florence is wonderful, minus the tourists. When I was able to travel my favorite thing to do was find the places and paths less traveled.
You are right about nationalist phrases, I am no "go USA" guy, and today especially with conflict everywhere...

but back to ietng as you said.

Doing a PR for the translations would be excellent as it makes things easier and less error prone. Two things. First do the PR against the branch not master. Second give me a heads up in case I am about to add or change translations.

I verified the zip issue. Another byproduct of the mbox extension. Added that for another user last minute, bad idea.
@cleidigh

**Comment by cleidigh** [comment]
@DenB10 
Not sure what you mean about the menus, they need unique ids.
If you mean the fact that the copyToClipboard functionality is in the msgCtx menu that's a bit of a competing interst, v115 versus old menu conflict thing.
Not sure how to satisfy everyone.
@cleidigh

**Comment by DenB10** [comment]
I meant that the "Copy to clipboard" entries were duplicated in the page context.
But after several attempts, I find that the "Copy to clipboard" commands built into "Export messages as..." don't work in the page context.

**Comment by cleidigh** [comment]
@DenB10 
Exactly, some users wanted the message menu in the display and this not only creates a conflict, but moves the menus one more level down.
I have to figure out a better way, but someone won't be happy.
@cleidigh

**Comment by DenB10** [comment]
>I have to figure out a better way, but someone won't be happy.
Yes, you have to make a choice, but I'm sure you'll make the right one ;-)

**Comment by DenB10** [comment]
@cleidigh 

I've made two PRs for the French translation: one for the "importexport-help.html" file and one for the "messages.json" file.

**Comment by cleidigh** [comment]
@DenB10 
ha ha, it will be my way or the highway!
Yes you are correct, one menu entry per extension. Also extensions at the bottom only %-{
@cleidigh

**Comment by Yarny0** [comment]
> Were both of your accounts pop type where you saw this failure?

Yes, mine was a POP3.

To be sure, I tried again, again with a fresh Thunderbird profile, complete default settings, IET beta version installed, also with default settings.  I created two mail-accounts: One with IMAP-connection, one with POP3-connection.  I sent two emails, so each account had one mail in its "Sent" folder and one mail in its "Inbox".

Exporting the IMAP account causes a warning by IET that not all messages might be present, but then worked flawlessly: It created an mbox-file for each folder.  Exporting the POP3 account does not show a warning, but just yields an empty folder.

**Comment by cleidigh** [comment]
@Yarny0 
@DenB10 
Found the TB issues with the pop account export.
Should be good now.
Also fixed the ascii option issue.
@cleidigh

**Comment by cleidigh** [comment]
@Yarny0 
Your testing and reporting is much appreciated especially as your main concern is eml export!

I have it on the b3 todo list to make the zip file unique. Adding the mbox extension option at the request of a user at the last minute was a mistake as  had long tentacles.

On the menus, this is something I have seen myself maybe twice. I am a little dubious of the menu api, however, I do have to see if there are any holes in my menu update functions.

And good year pop3 exports are working.
@cleidigh

**Comment by mforkel** [comment]
While it seems to work for other people, I couldn't manage to install import-export-tools-ng-14.0.1-b2-tb.xpi with the "Install Add-on from file..." command on the Add-ons management page of Thunderbird 115.3.3 (64 bit). I get a warning that the add-on couldn't be installed because it seems to be damaged. (Please note: This is my first attempt at installing a Thunderbird extension from a file, plus I'm on a German Thunderbird, so command name might differ.) Any suggestions?

**Comment by cleidigh** [comment]
@mforkel
Don't worry, nothing to do with German %-)
It's a weird Github issue. Here is a note on this : 
https://github.com/cleidigh/printing-tools-ng/issues/154

@cleidigh

**Comment by librelegal** [comment]
b3: Tested profile export under Windows10, worked fine  (~6,5 GB). Version 14 in contrary crashed Thunderbird

**Comment by cleidigh** [comment]
@librelegal 
Many thanks!
- Did you id the three files 
- When you did the exports, you had the mbox extension option checked?

I did a ton of testing on the naming, also did you restart? I may still have a cache issue 
@cleidigh

**Comment by librelegal** [comment]
Sorry, no I could not identify the three files that according to Windows were missing in the profile export in comparison to the original profile folder. 
 
Yesterday did not first check the mbox export option. I had simply chosen the inbox folder within TB, did a right click, chose to export the folder in the mbox format and then decided on the location. The file was called "inbox" and not "inbox.mbox".

![image](https://github.com/thundernest/import-export-tools-ng/assets/37673969/436ce88b-f2e5-4cd1-9276-c88d5bd6cd27)

Today with 115.4.1 and obviously after a restart on Windows I repeated the mbox export after I had first checked the mbox option and chosen the folder. The result was the same:  the file was called "inbox" and not "inbox.mbox". 

![image](https://github.com/thundernest/import-export-tools-ng/assets/37673969/e9ec2950-175c-439c-8e14-18fe5b83b06d)

**Comment by cleidigh** [comment]
@librelegal 
Very odd, I cannot reproduce.
Could you try exporting a folder to zip and capture the console?
This has some debug I am curious to see.
Unfortunately this mbox extension, simple as it should be is a pain...
Thanks 
@cleidigh

**Comment by Joergen** [comment]
Does it help to select "Use .mbox extension for mbox files (non-structured)" in the Misc tab?

**Comment by cleidigh** [comment]
@Joergen
Were you able to do an export with the mbox extension option and get an x.mbox result?
@cleidigh

**Comment by Joergen** [comment]
@cleidigh 
Yes - on a Linux computer.
Without a check mark next to "Use .mbox extension for mbox files (non-structured)" the add-on does not append the file extension to the filename.

**Comment by DenB10** [comment]
Hi everybody,
I agree with @Joergen
I'm on Windows and I don't have any problems with mbox export either.

**Comment by cleidigh** [comment]
@Joergen
Yes that is what the new option is for. The default is no extension because that is the Thunderbird standard (no mbox extension) therefore until I added the option IETNG has always followed Thunderbird.

So I have windows working with the extension, you have Linux working, but @librelegal is not getting the extension.
Perhaps we have another preferences conflict.
@cleidigh

**Comment by cleidigh** [comment]
@DenB10 
Excellent, so two platforms (three locales) provides confidence we have a configuration or update issue.
@cleidigh

**Comment by librelegal** [comment]
It DOES work on Windows, I do get a file called x.mbox. Truely sorry, I thought I only had to tick this

> ![image](https://user-images.githubusercontent.com/37673969/278216047-e9ec2950-175c-439c-8e14-18fe5b83b06d.png)

... but I also had to tick this:
![image](https://github.com/thundernest/import-export-tools-ng/assets/37673969/afe55475-279f-4177-a4ab-3fcb9c47124b)
(maybe it would be better to automatically tick this, when chosing to export a folder in that format?)

and 
![image](https://github.com/thundernest/import-export-tools-ng/assets/37673969/84b43a71-d948-41f4-98ee-f8263269aae8)

**Comment by librelegal** [comment]
And it does work on debian 11, KDE. Yippee Yeah

**Comment by librelegal** [comment]
> @librelegal Very odd, I cannot reproduce. Could you try exporting a folder to zip and capture the console? This has some debug I am curious to see. Unfortunately this mbox extension, simple as it should be is a pain... Thanks @cleidigh

And under Windows a .zip works, it does contain an .mbox

**Comment by cleidigh** [comment]
@librelegal 
Good, ok I was worried.
As I described to @Joergen above Thunderbird and therefore IETNG have always had NO extension so I am hesitant to change the default, however, when I make both structured imports and exports work with the mbox extension I will poll everyone to see if we have agreement.

@librelegal 
@DenB10 
@Joergen 
Also a related naming issue : 
The option Use only alphanumeric characters in filenames only allows a-z 0-9 and dash, all other characters replaced with underscores. I discovered even the period.
This may have made sense before, however, I have to change that.
@cleidigh

**Comment by cleidigh** [comment]
@librelegal 
Our posts crossed...
So to be clear, both Windows and Linux export with the extension as long as the option is checked.
So all is good?
@cleidigh

**Comment by librelegal** [comment]
> @librelegal Our posts crossed... So to be clear, both Windows and Linux export with the extension as long as the option is checked. So all is good? @cleidigh

Yep

**Comment by cleidigh** [comment]
@librelegal 
Super bueno!
Have to make sure one of best customers is cool %-} 
@cleidigh

**Comment by librelegal** [comment]
Smile. Here is a suggestion for the file name of these .mbox files, #486

**Comment by Yarny0** [comment]
For the record: I tested beta3 with a fresh profile, with mbox mail storage, with one IMAP and one POP3 account, with emails with and without attachment.

*   All export formats work flawlessly (I didn't test each function, but at least one export function for each format, and at least one attachment exporting function for each format that supports such).  In particular, I no longer can provoke IET to acccientally overwrite an existing file.
*   Import of multiple eml files and of a single mbox file work also flawlessly.
*   Full export of an account also works as expected.
    I no longer observed the wrong menu in the account's context menu -- I'm not sure if this is luck or is fixed (cf. https://github.com/thundernest/import-export-tools-ng/issues/466#issuecomment-1771305525).

This is a great version!

**Comment by DenB10** [comment]
I agree, all that remains to be done is to localize certain texts that are not yet translatable.

**Comment by cleidigh** [comment]
@DenB10 
My head is all over the place...
Do you know what translations are missing? I thought I had done them all.
@cleidigh

**Comment by DenB10** [comment]
@cleidigh
Unless I'm mistaken : 
- Lines 1974 and 1975 of the exportTools.js file ("Copy Headers to Clipboard" and "Multiple messages selected, only fist message headers copied to clipboard.") as well as lines 1903 and 1904 are not translated.
- In ImportMboxModule.js: lines 87 to 94 (not tested as I don't have a mbox larger than 4 GB).

**Comment by Mr-Update** [comment]
@cleidigh
Missing translation also exists in mboxImportExport.js und mboxImportExport-1.js (line 119):
- "This mbox exceeds the 4GB direct import size.\n\nDo you want to use the copy import method ?\n\nThis method will not do mbox processing.\nIf the mbox has not been processed, some messages may\nbe corrupted."

**Comment by cleidigh** [comment]
Translators:
@kiki-ja 
@DenB10 
@Mr-Update 
@Joergen 
If possible it would be much appreciated for these entity translations : 

"Copy Headers to Clipboard"
"Multiple messages selected, only fist message headers copied to clipboard."
"Copy Messages to Clipboard"
"Multiple messages selected, only the fist message copied to clipboard."

And I would like to thank you all again for making IETNG truly well localized!
@cleidigh

**Comment by Mr-Update** [comment]
@cleidigh
"Kopfzeilen in die Zwischenablage kopieren"
"Mehrere Nachrichten ausgewählt, nur die ersten Nachrichtenkopfzeilen in die Zwischenablage kopiert."
"Nachrichten in die Zwischenablage kopieren"
"Mehrere Nachrichten ausgewählt, nur die erste Nachricht in die Zwischenablage kopiert."

**Comment by DenB10** [comment]
@cleidigh

"Copy Headers to Clipboard" > "Copie des en-têtes du message dans le presse-papiers"
"Multiple messages selected, only fist message headers copied to clipboard." > "Plusieurs messages sont sélectionnés, seuls les en-têtes du premier seront copiés."
"Copy Messages to Clipboard" > "Copie du message dans le presse-papiers"
"Multiple messages selected, only the fist message copied to clipboard." > "Plusieurs messages sont sélectionnés, seul le premier sera copié."

>And I would like to thank you all again for making IETNG truly well localized!

You're welcome!

**Comment by kiki-ja** [comment]
@cleidigh

>If possible it would be much appreciated for these entity translations :

See Issue below for Japanese translation and update.

ja locale update · Issue #461 · thunderbird/import-export-tools-ng · GitHub
https://github.com/thunderbird/import-export-tools-ng/issues/461

@kiki-ja

**Comment by Joergen** [comment]
@cleidigh 

Here are the Danish translations:
"Copy Headers to Clipboard" -> "Kopier headers til udklipsholder"
"Multiple messages selected, only fist message headers copied to clipboard." -> "Flere meddelelser valgt, kun første meddelelses-header kopieret til udklipsholder."
"Copy Messages to Clipboard" -> "Kopier meddelelser til udklipsholder"
"Multiple messages selected, only the fist message copied to clipboard." -> "Flere meddelelser valgt, kun første meddelelse kopieret til udklipsholder."

I found a machine translated string that I missed the first time. If you could change the following string from the Danish **mboximport.dtd** file:
`<!ENTITY useMboxExt.label "Brug .mbox-udvidelsen til mbox-filer (ikke-struktureret)">` to `<!ENTITY useMboxExt.label "Brug filendelsen .mbox til mbox-filer (ikke-struktureret)">`

**Comment by cleidigh** [comment]
@kiki-ja 
@DenB10 
@Joergen 
@Mr-Update 
Thanks all.
Should have b4 out tomorrow and that should be good except if we want translation for release notes in help, thoughts?
@cleidigh

**Comment by Yarny0** [comment]
Hi, I tested all available export functions again.  Most of them work flawlessly, but I discovered the following issue:

When exporting messages (context menu after selecting two folded messages in the message list) as "New mbox file" from a POP3-account with "maildir" storage, no file is created, and the error console reports:

```
TypeError: uri is undefined   MailServices.jsm:20:17
```

I'm not sure if mbox export is supposed to be presented to the user here at all.  As I suspect this problem to be connected with the "wrong context menu in folder tree" bug (where only mbox export is offered for maildir folders, cf. https://github.com/thunderbird/import-export-tools-ng/issues/466#issuecomment-1771305525 ),
<details>
  <summary>I made precise notes on how I set up Thunderbird before I reached the broken export function</summary>

Most of the following is probably over-specific (like servernames or passwords), but I want to maximize reproducibility of the bug.

*   Start Thunderbird with a fresh profile
*   Go to "Add-Ons and Themes"
*   Disable "Update Add-Ons Automatically" in the gearwheel menu
*   Install `import-export-tools-ng-14.0.1-b4-tb.xpi` via "Install Add-On From File..." the gearwheel menu; acknowledge permissions with "Add"; acknowledge installation with "OK"
*   Go to Settings, "General" section
*   Set "Message Store Type for new accounts" to "... (maildir)"
*   Go back to the "Account Setup" tab
*   Create account:
    *   name `user1`, Email `user1@localhost`, Password `1`, "Remember" checked
    *   use "Configure manually"
    *   Incoming Server: "POP3", `localhost`, Port `110`, Security "None", Authentication "Normal pssword", User `user1`
    *   Outgoing Server: `localhost`, Port `25`, Security "None", "No authentication"
    *   "Done"
    *   Confirm the risks of unencrypted connections
    *   Note: user1 has two mails (each with one attachment) in its mailbox.  One message is a follow-up of the other, so Thunderbird shows these messages "folded".
*   close and restart thunderbird (Note: I'm doing this to create a copy of the profile folder to repeat tests without having to do the setup again)
*   click on user1's Inbox
*   right-click on the only folded message
*   select "Export Messages As...", "mbox Format", "New mbox File"
*   Enter path `/tmp/abc`, "Save"
</details>

Note that I also observed the "wrong context menu in folder tree" bug again, but alas, I still cannot reproduce it _sigh_ .

Thanks again for this add-on.

**Comment by DenB10** [comment]
Hi, there,

I don't have the same problem as @Yarny0 (my accounts are imap accounts) but I do have another problem:
The context menu of an account offers with version b4 two new items: "Copy Folder Path" and "Open Folder Directory" which do not work.

N.B.: thank you @cleidigh for allowing the help to be opened in a window, which makes it much easier (for me) to access the help.

**Comment by cleidigh** [comment]
@Yarny0 
@DenB10 
Once again kudos for all the testing help!

@Yarny0 
I have to mimic your setup (easy with your notes) for the mbox export. It's probably just a menu parameter thing, not too concerned.
The big one is showing the full folder context on the account.  I have seen this myself  so we are not crazy. I have tried all sorts of things and cannot reproduce either. I have a large and dynamic menu and I am weary of the api, but not sure how to debug yet.

@DenB10 
Yup I caught that AFTER posting b4. May remove them.
As far the help in a window , you must have been digging. You can thank @kiki-ja , it was his idea and next rev will get into options.
@cleidigh

**Comment by cleidigh** [comment]
@Yarny0 
I just found what might be a related issue.
My main test setup uses my imap test account and using my other addon LocalFolders I have multiple LFs one is setup as maildir.
I just found that if I right click on the maildir LF the imap account and its folders now show the full mailder menu.
So my question is do you always use maildir and or do you think the ctx errors were always with maildir. And, did the full ctx menu on the account show the full maildir or mbox menus.

I think I have something to work with because this state becomes sticky which is different than your observations and mine prior.
@cleidigh

**Comment by Yarny0** [comment]
> do you always use maildir

In my productive Thunderbird setups I always use maildir.  When testing IET, I try to mix storage types for more thorough testing.

> and or do you think the ctx errors were always with maildir.

No, it seemed to me that it depends on what menues are opened in which order.

I just tested again, this time only with one message and without attachment (also POP3/maildir account).  Context menu of the mail (not folder) offered mbox export (and eml/html/pdf/..), and mbox export of the message *works*.  Then I created an answer to that message so that thunderbird sees two messages which are displayed "folded".  Now, the mbox export *fails*.  Unfolding the messages and mbox-exporting each with its context menu *works*.  Folding them again, then using the mbox-export with the context menu of the folded message *fails*.

So I guess the issue is triggered by using the context menu on "folded" messages.

> And, did the full ctx menu on the account show the full maildir or mbox menus.

I am 85% sure that it only showed the mbox menus, i.e. a menu item with four submenu items.  I think I have never seen the maildir menu where I expected the mbox menu, but I have seen the mbox menu where I expected the maildir menu.

I might be wrong.

**Comment by cleidigh** [comment]
@Yarny0 
I can reproduce the folded messages issue, however, with one specific aspect:

First select a single non-folded message then right click on a folded set and try any message export and you get only the first of the folded messages.
If I directly only select the folded message I get the thread for new mbox or other types.
Does that match your observations?
I have to redo the selection handling.
@cleidigh

**Comment by Yarny0** [comment]
> First select a single non-folded message then right click on a folded set and try any message export and you get only the first of the folded messages.

Not exactly, it seems to depend on which message is selected (blue background).  I tried this (POP3/maildir) with three messages in the inbox, two of them folded, the third one being unrelated to the other messages:

* left-click on the single message
* right-click on the folded messages, select eml or mbox export
* result: the single unfolded message gets exported

Then I tried this:

* left-click on the single message
* left-click with ctrl on it again, so that *no* message is selected
* right-click on the folded messages, select eml or mbox export
* result: nothing gets exported, the error console shows the type error

I conclude that IET always works on selected messages.  If no message is selected, it fails due to the lack of selected messages.  In my experiment described in https://github.com/thunderbird/import-export-tools-ng/issues/466#issuecomment-1792896562 , I apparently did not select any messages after starting thunderbird and went directly to the message list context menu to export something; in that case, nothing is selected.

Thinking loud (of couse you don't have to follow my thoughts): From a user experience perspective, I assume it is not unreasonable to work on selected messages only.  I'm not sure what I (or any user) would expect from selecting message A and then right-clicking on message B to initiate an export.  In any case, the menu item should indicate what is about to happen ("export *selected* messages" or similar).

> If I directly only select the folded message I get the thread for new mbox or other types.

I'm not sure if I understand that question.
In the situation described above (three messages, two folded, POP3/maildir), the context menu in the message list always looks the same, irrespective of which messages are un/folded or de/selected: It shows the options EML (submenu), HTML (submenu), PDF, plain text (submenu), CSV, mbox (submenu).

**Comment by cleidigh** [comment]
@Yarny0 
All
I completely redesigned the selected messages handling now that I understand better what was going on. Essentially I guarantee to use the wext side to handle the selected messages except for selections over 100.
All the quirks get handled this way.

@cleidigh

**Comment by cleidigh** [comment]
Translators: 
@kiki-ja
@DenB10 
@Mr-Update 
@Joergen 

We are there!! B5 has the last patch.
I am going to add the release notes to the top of help.
I can auto translate so if people don't have time, no big deal.

 
Release Notes: v14.0.1

- New: Add Account level mbox import
- New: Add Account level mbox flattened export
- New: Support for mbox import and export to filesys levels
- Fix CSV export to text #451, #463
- Fix export folder to ZIP
- Fix help load only on IETNG updates
- Add preference to never load help on updates #458
  extensions.importexporttoolsng.export.help.showOnInstallAndUpdate
- Fix help for sublocales and unsupported locales
- Fix mbox From_ separators to adhere to RFC 4155 #455
- Add back Export Messages in message display windows #459
- Fix Account export for pop3
- Fix buffer boundary From_ escaping
- Fix selected messages handling #485
- Complete de locale update @Mr-Update
- Complete fr locale update @DenB10
- Complete da locale update @Joergen
- Updated ja locale @kiki-ja

Thanks as always 
@cleidigh

**Comment by DenB10** [comment]
@cleidigh 

>I am going to add the release notes to the top of help.

Not knowing exactly where to insert the release notes in the help file, I'm providing the raw French translation below:

```
Notes de version 14.0.1
- Nouveau : ajout de l’importation mbox au niveau du compte
- Nouveau : ajout de l’exportation mbox à plat au niveau du compte
- Nouveau : prise en charge de l’importation et de l’exportation mbox au niveau du système de fichiers
- Correction de l’exportation CSV en texte (#451, #463)
- Correction de l’exportation des dossiers en ZIP
- Correction : chargement de l’aide uniquement lors d’une mise à jour de IETNG
- Ajout de la préférence « extensions.importexporttoolsng.export.help.showOnInstallAndUpdate »
  permettant de ne jamais charger l'aide lors d’une mise à jour (#458)
- Correction : aide pour les sous-langues et les langues non supportées
- Correction : séparateurs mbox « From_ » afin de respecter RFC 4155 (#455)
- Réintégration du menu « Exporter les messages » dans le panneau d’affichage des messages (#459)
- Correction de l’exportation d’un compte pop3
- Correction : échappement de « From_ » en limite de la zone tampon
- Correction de la gestion des messages sélectionnés (#485)
- Mise à jour complète de la locale « de » (@Mr-Update)
- Mise à jour complète de la locale « fr» (@DenB10)
- Mise à jour complète de la locale « da » (@Joergen)
- Mise à jour de la locale « ja » (@kiki-ja)

```
I also have a practical request before the final version is released: 

Is it possible that the options window is no longer "modal"?
When you call up help from the options page (via the question mark icons), the help is opened in a tab and cannot be accessed because it is blocked by the options window, which remains in the foreground.
(I hope you've understood what I mean?)

**Comment by cleidigh** [comment]
@librelegal 
In b5 I reimplemented the handling of selected messages to deal with issues involved with folded threads and a couple other items. This effects the exportSelectedMessages() interface you use with shortcuts. I tested with IETNG shortcuts. If you have a moment to check I want to make sure you are still good.
Also note you have been added to the infamous IETNG crowd in help(rel version pending)
- #466

cheers
@cleidigh

**Comment by cleidigh** [comment]
@DenB10 
Merci...
All integrated, you can see in b6.
Almost ready for release.
https://github.com/thunderbird/import-export-tools-ng/blob/v14.0.1/xpi/beta/import-export-tools-ng-14.0.1-b6-tb.xpi
@cleidigh

**Comment by librelegal** [comment]
> @librelegal In b5 I reimplemented the handling of selected messages to deal with issues involved with folded threads and a couple other items. This effects the exportSelectedMessages() interface you use with shortcuts. I tested with IETNG shortcuts. If you have a moment to check I want to make sure you are still good. Also note you have been added to the infamous IETNG crowd in help(rel version pending)
> 
>     * [v14.0.1 Release Status - Beta Download #466](https://github.com/thunderbird/import-export-tools-ng/issues/466)

Dear Christopher, the exportSelectedMessages() thing works with B6. I thank you a lot for also respecting that need.

**Comment by kiki-ja** [comment]
@DenB10

>When you call up help from the options page (via the question mark icons), the help is opened in a tab and cannot be accessed because it is blocked by the options window, which remains in the foreground.

ImportExportTools NG 14.0.1-b５, b6
Thunderbird 115.4.2
Windows 10 Pro 22H2

Workaround Method: How to Open Help in a Window
1. Select an account, folder, message, or the message pane.
2. Right mouse click.
3. Click Help on the context menu.

Caution:
This method may be changed in future versions.

@kiki-ja

**Comment by DenB10** [comment]
@kiki-ja 

> Workaround Method: How to Open Help in a Window
> 1. Select an account, folder, message, or the message pane.
> 2. Right mouse click.
> 3. Click Help on the context menu.

Yes, I know that, but, as I was saying, it doesn't work if you call up help from the options page by clicking on a question mark icon: help then opens in a tab (as when IENG is updated) at the relevant heading level rather than a window, and as the options window remains in the foreground, it hides the help, which I find annoying.
You then have to move the options window to read the help, or close the options window and then reopen it to change an option, which isn't very practical, is it?
I found a workaround by deleting "modal" on line 1593 of the mboximport.js file but, as I'm not a developer, I'm not sure that's right! 
However, it works: the options window doesn't stay in the foreground and you can consult the help comfortably without having to close the options window.

@DenB10

**Comment by kiki-ja** [comment]
@DenB10

I understand what you are saying.
Perhaps the addition of a debug option to open in a window, similar to PrintingTools NG, would help.

In the case of my PrintingTools NG:
Since my PC display is not high resolution (FHD:1902x1080), I modified the width of the PTNG options window to be a little smaller and the PTNG help window in the same way so that both windows can be edited side by side.

![sample](https://github.com/thunderbird/import-export-tools-ng/assets/80428201/e4975403-947a-4c45-aa5b-3456f098bcc0)

@kiki-ja

**Comment by DenB10** [comment]
@kiki-ja 
>Perhaps the addition of a debug option to open in a window, similar to PrintingTools NG, would help.

You're teaching me something: I didn't know there was a debug option to open help in a window in PTNG, what is this way?

**Comment by cleidigh** [comment]
@DenB10 
@kiki-ja 
Opening the options as non modal is indeed not a problem, it's the unintended consequences.
For instance the thing is I would have to add detection of the options already open and prevent someone opening another dialog.  I have not imagined other things, but there may be others.

As @kiki-ja mentioned, there is a debug option in PTNG at his suggestion to enable opening help in a separate window instead of a tab. I didn't get it into the UI and should have for v3.0.1
The debug option is helpinwin which you place in Debug Options

Short term I can do a preference for this then do the UI and translations in the next release. 
The question is what is the best way going forward and can we or should we do this before releasing v14.0.1?
@cleidigh

**Comment by DenB10** [comment]
@cleidigh
>Opening the options as non modal is indeed not a problem, it's the unintended consequences.
>For instance the thing is I would have to add detection of the options already open and prevent someone opening another dialog. >Ihave not imagined other things, but there may be others.

In fact, I don't understand why IETNG's options window should be modal when PTNG's isn't!

>As @kiki-ja mentioned, there is a debug option in PTNG at his suggestion to enable opening help in a separate window instead >of a tab. I didn't get it into the UI and should have for v3.0.1
>The debug option is helpinwin which you place in Debug Options

"I must have shit in my eyes" (French expression) because this debug option is mentioned in the PTNG help. Anyway, thanks for the info.

>Short term I can do a preference for this then do the UI and translations in the next release.
>The question is what is the best way going forward and can we or should we do this before releasing v14.0.1?

Don't worry, this is not a major concern and can wait. You've already done a lot of work on version 14.0.1 and I think (like you, I believe) it's time to release it.

PS: Did you read my question in my last PR comment about updating the French help?

**Comment by Yarny0** [comment]
For the record: I tested all export methods/formats with a POP3/maildir and with a IMAP/maildir account.  I didn't look into each exported file, but verified that their file size was approximately what was to be expected.  Everything worked flawlessly.  I also tested how export works with folded/unfolded messages and with only some messages selected; this also worked flawlessly: I always exported what I expected it to export, and I could no longer reproduce the problem where it effectively exported nothing.

At the end of my experiments, I again had the mbox-4-items-export-menu in the context menu of my (maildir) inbox folders, but again I found no way to reliably reproduce it.

Thanks for your work!

**Comment by cleidigh** [comment]
@DenB10 
So I figured out the way to check if an option window is open and then realized I can avoid opening a second window if a user attempts AND then I focus on the already open window. Let me zip a pre7 and you can try to break it.
Give me five...
Yes I missed removing the export remote entry in help.
@cleidigh

**Comment by cleidigh** [comment]
@DenB10 
Ok here is the zip, you can install like an xpi.
So if this doesn't break things we can do it. I may just do the window option as well.
I need to delay to look at profile import export anyway.
[import-export-tools-ng-14.0.1-b7pre1-tb.zip](https://github.com/thunderbird/import-export-tools-ng/files/13311722/import-export-tools-ng-14.0.1-b7pre1-tb.zip)

@cleidigh

**Comment by cleidigh** [comment]
@librelegal 
Excellent, thank you.
The new selection handling should be correct for all scenarios now.
@cleidigh

**Comment by cleidigh** [comment]
@Yarny0 
Great, the selection handling should be spot on now.
The menu issue is perplexing, I cannot figure that one out yet..
Btw just doing last minute change to make options non modal to view help better. See up thread if interested.
@cleidigh

**Comment by DenB10** [comment]
@cleidigh
Wow, what responsiveness!
After a few tries, I've had no problems whatsoever.
You're a boss!

**Comment by cleidigh** [comment]
@DenB10 
Cool, so far I can't think of anything other than preventing duplicate windows so hopefully we are good.
For the window option I need the translation for:
"Open help in a window instead of a tab"
TheBoss

**Comment by DenB10** [comment]
@TheBoss
"Open help in a window instead of a tab" > "Ouvrir l’aide dans une fenêtre et non un onglet"

**Comment by cleidigh** [comment]
@DenB10 
Merci mon frere
Q: have you done anything with profile import export?
Have a user I think is trying to import old one. Don't think I handle well in ui information.
That's my next investigation.
@cleidigh

**Comment by DenB10** [comment]
@cleidigh 
In fact, I don't use IETNG's profile import-export function. I always use the same profile, which I manually save on a regular basis. If necessary, I restore a previous backup, taking care to clear Thunderbird's cache. I don't need several profiles either.

However, I've just done a few quick tests and it seems to work, except for a few quirks I can't explain: when I import a previously exported profile, traces of the previous profile seem to remain, even after restarting Thunderbird and clearing its cache. Maybe I'll tell you more tomorrow if you ask me to investigate further. But then again, I'm not really comfortable with this IETNG feature.

**Comment by cleidigh** [comment]
@Yarny0 
While working on the help stuff I realized I had an error in the preference I added for you. The preference was: 
```
extensions.importexporttoolsng.export.help.showOnInstallAndUpdate

should be, and will change for the release as of b7

extensions.importexporttoolsng.help.showOnInstallAndUpdate
```
Hopefully this is not to inconvenient, but better to change now.
@cleidigh

**Comment by sembuk** [comment]
I usually use Thunderbird with 4 windows opened. "Select the destination directory" dialog opens 4 times when I exporting emails by selecting target directory and clicking save button. This behaviour happens also on importing, opening "Options" from right-click menu.
My specs:

- Windows 10 Pro 22H2
- Thunderbird 115.4.2 (64-bit)
- ImportExportTools NG v14.0.1-b7

**Comment by cleidigh** [comment]
@sembuk 
Thank you for catching this!
Never using multiple windows I failed to contemplate or test this.
I just posted b8 with the patch. Both the export and import, really all commands were affected. I tested with multiple windows, both the export and import.
Hopefully you have a moment to verify. I would like to post to the review Q tomorrow.
Thanks again 
@cleidigh 
- #466

**Comment by librelegal** [comment]
> @librelegal Excellent, thank you. The new selection handling should be correct for all scenarios now. @cleidigh

the save message to desktop function works in b8, thank you so much

**Comment by cleidigh** [comment]
@librelegal 
Thank you as always.
I am posting v14.0.1 to the review Q today!
This was certainly a big maintenance release %-}
@cleidigh

**Comment by sembuk** [comment]
> @sembuk Thank you for catching this! Never using multiple windows I failed to contemplate or test this. I just posted b8 with the patch. Both the export and import, really all commands were affected. I tested with multiple windows, both the export and import. Hopefully you have a moment to verify. I would like to post to the review Q tomorrow. Thanks again @cleidigh
> 
> * [v14.0.1 Release Status - Beta Download #466](https://github.com/thunderbird/import-export-tools-ng/issues/466)

Great, Works great, thank you for fixing this.
